server/routes/admins.ts

```diff
@@ -7,10 +7,16 @@ import { logError } from '../logger.ts'
 
 const router = express.Router()
 
+// Get and view all the tasks that have been assigned to a client by an admin by that clients id
 router.get('/:clientId/tasks', validateAccessToken, async (req, res) => {
   const adminId = req.auth?.payload.sub
   const clientId = req.params.clientId
 
+  if (!adminId) {
+    res.status(400).json({ message: 'Please login with your admin Id' })
+    return
+  }
+
   try {
     const result = await getTasksByAdmin(adminId as string, clientId)
     if (!result) {
```

Comment:
for the error 404 test, check how this line is handling the empty array, maybe its not undefined?